keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -293,10 +295,22 @@ def generate(
                 "`preprocessor` is set before calling `generate`."
             )
         sampler = keras_nlp.samplers.get(sampler)
-        # `jit_compile` is a property of keras.Model after tf 2.12.
-        # Use `getattr()` for backwards compatibility.
-        sampler.jit_compile = getattr(self, "jit_compile", True)
+        if sampler.__class__.__name__ == "BeamSampler":
```

Comment:
Can't you just direct compare `if sampler.__class__ == BeamSampler` here? Better to do this outside string land I think